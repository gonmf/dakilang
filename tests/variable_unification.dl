alpha("hello", "world").
beta(A, B) :- alpha(A, B).
charlie(900, 900).
delta(A, B) :- charlie(A, B).

beta(A, A)? # No
delta(A, B)? # A = 900, B = 900
delta(A, A)? # A = 900
delta(900, 900)? # Yes
delta(900, A)? # A = 900

abc(123).
abc(123)?
