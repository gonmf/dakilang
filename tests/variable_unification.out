> alpha("hello", "world").
> beta(A, B) :- alpha(A, B).
> charlie(900, 900).
> delta(A, B) :- charlie(A, B).
>
> beta(A, A)? # No
No

> delta(A, B)? # A = 900, B = 900
A = 900
B = 900

> delta(A, A)? # A = 900
A = 900

> delta(900, 900)? # Yes
Yes

> delta(900, A)? # A = 900
A = 900

>
> abc(123).
> abc(123)?
Yes

